### PR TITLE
Fixed codegen cyclic dependency given table 'Table' 

### PR DESCRIPTION
**Pull request by trevorsibanda**
#1486

### CODE REVIEW

File: slick-testkit/src/codegen/scala/slick/test/codegen/GenerateRoundtripSources.scala [hunk 1]
```diff
@@ -65,6 +65,16 @@ class Tables(val profile: JdbcProfile){
   }
   val SingleNonOptionColumn = TableQuery[SingleNonOptionColumn]
 
+  /** Tests table name cyclic dependency */
+  case class ColumnRow(id: Long , name: String  )
+  class Column(tag: Tag) extends profile.api.Table[ColumnRow](tag, "column") {
```
Comment (lastland) [inline slick-testkit/src/codegen/scala/slick/test/codegen/GenerateRoundtripSources.scala]: What is the `Column` for here? The creation of table `Table` in `h2.sql` alone would not suffice?

Comment (trevorsibanda) [inline slick-testkit/src/codegen/scala/slick/test/codegen/GenerateRoundtripSources.scala]: You're right. I should remove it.

**Comment by cvogt** [comment]
LGTM!

**Comment by cvogt** [comment]
lets just squash it

File: slick-testkit/src/codegen/scala/slick/test/codegen/GenerateRoundtripSources.scala [hunk 2]
```diff
@@ -65,6 +65,8 @@ class Tables(val profile: JdbcProfile){
   }
   val SingleNonOptionColumn = TableQuery[SingleNonOptionColumn]
 
+
+
```
Comment (lastland) [inline slick-testkit/src/codegen/scala/slick/test/codegen/GenerateRoundtripSources.scala]: This is not good. :wink:

Comment (cvogt) [inline slick-testkit/src/codegen/scala/slick/test/codegen/GenerateRoundtripSources.scala]: yeah. whitespace changes should be avoided to reduce history noise :)

File: slick-testkit/src/codegen/resources/dbs/h2.sql [hunk 3]
```diff
@@ -25,3 +25,9 @@ CREATE TABLE "PERSON"(
   "ID" NUMBER(11,0) NOT NULL,
   "PERSON_TYPE" CHAR(1) DEFAULT 'Y' NOT NULL
 );
+
+CREATE TABLE "Table" (
+	"ID" NUMBER(11,0) NOT NULL,
```
Comment (lastland) [inline slick-testkit/src/codegen/resources/dbs/h2.sql]: Are the next two columns necessary?

Comment (trevorsibanda) [inline slick-testkit/src/codegen/resources/dbs/h2.sql]: No they're not necessary, test was on table name

**Pull request merged**